eclipse-collections/src/main/java/org/eclipse/collections/impl/map/mutable/UnifiedMap.java

```diff
@@ -569,6 +571,90 @@ private <P> V chainedUpdateValueWith(
         return result;
     }
 
+    @Override
+    public V merge(K key, V value, BiFunction<? super V, ? super V, ? extends V> remappingFunction)
+    {
+        Objects.requireNonNull(value, "value cannot be null");
+        Objects.requireNonNull(remappingFunction, "remappingFunction cannot be null");
+
+        int index = this.index(key);
+        Object cur = this.table[index];
+        if (cur == null)
+        {
+            this.table[index] = UnifiedMap.toSentinelIfNull(key);
+            this.table[index + 1] = value;
+            ++this.occupied;
+            return value;
+        }
+        if (cur != CHAINED_KEY && this.nonNullTableObjectEquals(cur, key))
+        {
+            V newValue = remappingFunction.apply((V) this.table[index + 1], value);
+            this.table[index + 1] = newValue;
```

Comment:
this doesn't look right. if `newValue` is null, we must clear the key slot.

Reply:
Good point, will update

Reply:
Fixed this, and stepped through the tests in the debugger to make sure the new lines are hit in tests.